test/models/bot/smooch_7_test.rb

```diff
@@ -660,4 +660,13 @@ def teardown
       end
     end
   end
+
+  test "should replace message_id placeholder" do
+    uid = random_string
+    tm = create_tipline_message uid: uid
+    text = 'Foo {{message_id}} bar {{message_id}}'
+    external_id = tm.external_id
+    output = Bot::Smooch.replace_placeholders(uid, text)
+    assert_equal "Foo #{external_id} bar #{external_id}", output
+  end
```

Comment:
For the test, I suggest you set an explicit `external_id` and assert it at the end, for example:

```ruby
tm = create_tipline_message uid: uid, external_id: 'abc123'
[...]
assert_equal "Foo abc123 bar abc123", output
```

Reply:
Done